### PR TITLE
Added setStdIoUnbuffered for unbuffered stdio (fixes #2146).

### LINKED ISSUE #2146: Feature request: unbuffered I/O
Python offers a command-line argument `-u` that offers an unbuffered version of stdio.  This is extremely useful in some circumstances.  Would it be possible to add this feature to Nim such that the compiled executables use unbuffered stdio?

### PR DESCRIPTION
**Pull request by abudden**

### CODE REVIEW

**Comment by Araq** [comment]
Failing with: 

```
$ nim e install_nimble.nims
Hint: used config file '/home/travis/build/nim-lang/Nim/config/nim.cfg' [Conf]
lib/system/sysio.nim(358, 19) Error: undeclared identifier: 'stdout'
```

**Comment by abudden** [comment]
> Failing with:
> ```
> $ nim e install_nimble.nims
> Hint: used config file '/home/travis/build/nim-lang/Nim/config/nim.cfg' [Conf]
> lib/system/sysio.nim(358, 19) Error: undeclared identifier: 'stdout'
> ```

Hopefully the new version should fix that.

File: lib/system.nim [hunk 1]
```diff
@@ -2715,6 +2715,9 @@ when not defined(JS): #and not defined(nimscript):
       ##
       ## Default mode is readonly. Returns true iff the file could be reopened.
 
+    proc setStdIoUnbuffered() {.tags: [], benign.}
```
Comment (Araq) [inline lib/system.nim]: Export marker missing.

Comment (abudden) [inline lib/system.nim]: Fixed.

**Pull request merged**